cardinal/ecs/component.go

```diff
@@ -213,3 +214,114 @@ func WithDefault[T any](defaultVal T) ComponentOption[T] {
 		c.validateDefaultVal()
 	}
 }
+
+// Get returns component data from the entity.
+func GetComponent[T component.IAbstractComponent](w *World, id entity.ID) (comp *T, err error) {
+	var t T
+	name := t.Name()
+	c, ok := w.nameToComponent[name]
+	if !ok {
+		return nil, errors.New("Must register component")
+	}
+	value, err := w.StoreManager().GetComponentForEntity(c, id)
+	if err != nil {
+		return nil, err
+	}
+	t, ok = value.(T)
+	if !ok {
+		comp, ok = value.(*T)
+		if !ok {
+			return nil, fmt.Errorf("type assertion for component failed: %v to %v", value, c)
+		}
+	} else {
+		comp = &t
+	}
+
+	return comp, nil
+}
+
+// Set sets component data to the entity.
+func SetComponent[T component.IAbstractComponent](w *World, id entity.ID, component *T) error {
+	var t T
+	name := t.Name()
+	c, ok := w.nameToComponent[name]
+	if !ok {
+		return fmt.Errorf("%s is not registered, please register it before updating", t.Name())
+	}
+	err := w.StoreManager().SetComponentForEntity(c, id, component)
+	if err != nil {
+		return err
+	}
+	w.Logger.Debug().
+		Str("entity_id", strconv.FormatUint(uint64(id), 10)).
+		Str("component_name", c.Name()).
+		Int("component_id", int(c.ID())).
+		Msg("entity updated")
+	return nil
+}
+
+func UpdateComponent[T component.IAbstractComponent](w *World, id entity.ID, fn func(*T) *T) error {
+	var t T
+	name := t.Name()
+	c, ok := w.nameToComponent[name]
+	if !ok {
+		return fmt.Errorf("%s is not registered, please register it before updating", t.Name())
+	}
+	if _, ok := w.nameToComponent[c.Name()]; !ok {
+		return fmt.Errorf("%s is not registered, please register it before updating", c.Name())
+	}
+	val, err := GetComponent[T](w, id)
+	if err != nil {
+		return err
+	}
+	updatedVal := fn(val)
+	return SetComponent[T](w, id, updatedVal)
+}
+
+//// EachComponent iterates over the entities that have the component.
```

Comment:
is this commented block being addressed in another PR? i think we're OK with removing component queries and requiring the user to use the query package

Reply:
Yeah this comment block will be addressed in another PR. As i address these issues in each PR I will remove the related comment.